src/app-logic/url-handling.js

```diff
@@ -727,11 +727,11 @@ function convertHiddenLocalTracksByPidFromString(
     if (!stringPart.includes('-')) {
       continue;
     }
-    const pidString = stringPart.slice(0, stringPart.indexOf('-'));
-    const hiddenTracksString = stringPart.slice(pidString.length + 1);
-    const pid = Number(pidString);
+    // TODO: handle escaped dashes and tildes in pid strings
```

Comment:
Could you also file a bug for this?

Reply:
Filed #4512 and mentioned it in the comments.